File: python/2000.py
# Link: https://leetcode.com/problems/reverse-prefix-of-word/description/
# Runtime: 31 ms
# Memory: 16.55 MB

class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        index_of_ch = word.find(ch)
        if index_of_ch != -1:
            return word[index_of_ch::-1] + word[index_of_ch+1:]

        return word

File: c++/2000.cpp
// Link: https://leetcode.com/problems/reverse-prefix-of-word/description/
// Runtime: 0 ms
// Memory: 7.48 MB

class Solution 
{
public:
    std::string reversePrefix(const std::string& word, char ch) 
    {
        std::string reversedString = word;
        short indexOfCh = 0;
        while ((indexOfCh < static_cast<short>(reversedString.size())) && (reversedString[indexOfCh] != ch))
            indexOfCh++;
        
        if (indexOfCh != static_cast<short>(reversedString.size()))
            for (int i = 0; i < indexOfCh / 2 + 1; i++)
                std::swap(reversedString[i], reversedString[indexOfCh - i]);

        return reversedString;
    }
};
